inst/js/run-r/btw-run-r.js

```diff
@@ -0,0 +1,193 @@
+  }
+
+  /**
+   * Toggle the collapsed/expanded state
+   * @param {Event} e
+   */
+  toggleCollapse(e) {
+    e.preventDefault()
+    this.expanded = !this.expanded
+    this.render()
+  }
+
+  /**
+   * Formats the title for display in the card header. Uses the `titleTemplate`,
+   * replacing `{title}` with the actual title or name of the tool.
+   * @returns {string}
+   */
+  formatTitle() {
+    const displayTitle = `<span class="tool-title-name">${
+      this.toolTitle || "Run R Code"
+    }</span>`
+    return this.titleTemplate.replace("{title}", displayTitle)
+  }
+
+  /**
+   * Render the component
+   */
+  render() {
+    const requestId = this.getAttribute("request-id") || "unknown"
+    const code = this.getAttribute("code") || ""
+    const headerId = `tool-header-${requestId}`
+    const contentId = `tool-content-${requestId}`
+
+    // Get the output HTML from child content (set during initial render)
+    const outputHtml = this._outputHtml || this.innerHTML
+    this._outputHtml = outputHtml
+
+    const collapsedClass = this.expanded ? "" : " collapsed"
+
+    this.innerHTML = `
+      <div class="shiny-tool-card card bslib-card html-fill-item html-fill-container m-0">
+        <button
+          class="card-header"
+          id="${headerId}"
+          aria-expanded="${this.expanded}"
+          aria-controls="${contentId}"
+        >
+          <div class="tool-icon ${this.classStatus}">${this.icon}</div>
+          <div class="tool-title ${
+            this.classStatus
+          }">${this.formatTitle()}</div>
+          <div class="tool-spacer"></div>
+          <div class="collapse-indicator">${ICONS.plus}</div>
+        </button>
+        <div
+          class="card-body bslib-gap-spacing html-fill-item html-fill-container${collapsedClass}"
+          id="${contentId}"
+          role="region"
+          aria-labelledby="${headerId}"
+          ${!this.expanded ? 'inert=""' : ""}
+        >
+          <div class="btw-run-source">
+            <shiny-markdown-stream
+              content="${this.escapeAttr(markdownCodeBlock(code, "r"))}"
+              content-type="markdown"
+            ></shiny-markdown-stream>
+          </div>
+          <div class="btw-run-output">
+            ${outputHtml}
+          </div>
+        </div>
+      </div>
+    `
+
+    // Add click handler to header
+    const header = this.querySelector(".card-header")
+    if (header) {
+      header.addEventListener("click", (e) => this.toggleCollapse(e))
+    }
+  }
+
+  /**
+   * Escape a string for use in an HTML attribute
+   * @param {string} str
+   * @returns {string}
+   */
+  escapeAttr(str) {
+    return str
+      .replace(/&/g, "&amp;")
+      .replace(/"/g, "&quot;")
+      .replace(/'/g, "&#39;")
+      .replace(/</g, "&lt;")
+      .replace(/>/g, "&gt;")
+  }
+}
+
+// Register the custom element
+customElements.define("btw-run-r-result", BtwRunRResult)
```

Comment:
We should be a little more careful and only register the element if not already defined